File: das/src/main/java/com/ctrip/sysdev/das/domain/RequestMessage.java
package com.ctrip.sysdev.das.domain;

import java.util.List;

import com.ctrip.sysdev.das.domain.enums.OperationType;
import com.ctrip.sysdev.das.domain.enums.StatementType;

/****
 * 
 * @author gawu
 * 
 */
public class RequestMessage {
	private String dbName;

	private StatementType statementType; // always

	private OperationType operationType; // always

	private boolean useCache; // always

	private String spName;

	private String sql;

	private List<StatementParameter> args;// always

	private int flags; // always

	
	public String getDbName() {
		return dbName;
	}

	public void setDbName(String dbName) {
		this.dbName = dbName;
	}

	public StatementType getStatementType() {
		return statementType;
	}

	public void setStatementType(StatementType statementType) {
		this.statementType = statementType;
	}

	public OperationType getOperationType() {
		return operationType;
	}

	public void setOperationType(OperationType operationType) {
		this.operationType = operationType;
	}

	public boolean isUseCache() {
		return useCache;
	}

	public void setUseCache(boolean useCache) {
		this.useCache = useCache;
	}

	public String getSpName() {
		return spName;
	}

	public void setSpName(String spName) {
		this.spName = spName;
	}

	public String getSql() {
		return sql;
	}

	public void setSql(String sql) {
		this.sql = sql;
	}

	public List<StatementParameter> getArgs() {
		return args;
	}

	public void setArgs(List<StatementParameter> args) {
		this.args = args;
	}

	public int getFlags() {
		return flags;
	}

	public void setFlags(int flags) {
		this.flags = flags;
	}

	/****
	 * 
	 * @return
	 */
	public int propertyCount() {
		return 6;
	}

}

File: dao_cs/platform/platform.dao/request/RequestMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using platform.dao.enums;
using platform.dao.param;

namespace platform.dao.request
{
    public class RequestMessage
    {
        /// <summary>
        /// SQL指令类型，SP或SQL
        /// </summary>
        public StatementType StatementType { get; set; }

        /// <summary>
        /// 操作类型，读写分离使用
        /// </summary>
        public OperationType OperationType { get; set; }

        /// <summary>
        /// 是否使用缓存
        /// </summary>
        public bool UseCache { get; set; }

        /// <summary>
        /// SP名
        /// </summary>
        public string SpName { get; set; }

        /// <summary>
        /// SQL语句
        /// </summary>
        public string Sql { get; set; }

        /// <summary>
        /// 参数
        /// </summary>
        public StatementParameterCollection Parameters { get; set; }

        /// <summary>
        /// 标识
        /// </summary>
        public int Flags { get; set; }


        public int GetPropertyCount()
        {
            return 6;
        }

    }
}
